src/main/java/com/google/api/codegen/transformer/csharp/CSharpModelTypeNameConverter.java

```diff
@@ -244,6 +239,16 @@ private TypeName getTypeNameForTypedResourceName(
 
   @Override
   public TypedValue getEnumValue(TypeRef type, String value) {
-    throw new UnsupportedOperationException("getEnumValue not supported by C#");
+    for (EnumValue enumValue : type.getEnumType().getValues()) {
+      if (enumValue.getSimpleName().equals(value)) {
```

Comment:
The exactly same pattern is seen among all languages. Also, `getEnumValue` is only referred from `ModelTypeTable`.

I think it's better to:
- extract `for...if...throw` code block into `ModelTypeTable`
- update `getEnumValue` interface as `public TypedValue getEnumValue(TypeRef type, EnumValue value)`

This will be beneficial for getting zero value too, it's gonna be `getEnumValue(type, type.getEnumType().getValues().get(0))`.